tests/algorithms/test_algorithm_resumption.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+import os
+import textwrap
+from typing import Optional, Type, Tuple
+
+import pytest
+
+from composer.algorithms import AlgorithmHparams
+from composer.callbacks.checkpoint_saver import CheckpointSaver
+from composer.trainer.trainer_hparams import algorithms_registry
+from composer.core.precision import Precision
+from composer.core.time import Time, TimeUnit
+from composer.datasets import SyntheticHparamsMixin
+from composer.trainer.devices import CPUDeviceHparams, DeviceHparams, GPUDeviceHparams
+from composer.trainer.trainer_hparams import TrainerHparams
+from composer.utils import run_directory
+from composer.utils.checkpoint import _is_archive
+from tests.fixtures.dummy_fixtures import dummy_num_classes
+from tests.trainer.test_checkpoint import assert_checkpoints_equivalent
+
+
+algo_hparams_overrides = {
+    "swa": {
+        "swa_start": "0.4dur",
+        "swa_end": "0.8dur"
+    },
+    "cutmix": {
+        "num_classes": dummy_num_classes
+    },
+    "mixup": {
+        "num_classes": dummy_num_classes
+    }    
+}
+
+simple_net_skiplist = {
```

Comment:
perhaps just have one skip list:
```
skip_list = {
    'default': {'colout': ...}
    'resnet50_synthetic': {, ..},
}
```

Then below in your code you can do this once:
```
if algo in skip_list[model_name]:
```
```